scanpy/readwrite.py

```diff
@@ -379,10 +379,22 @@ def read_visium(
             files['scalefactors_json_file'].read_bytes()
         )
 
-        adata.uns["spatial"][library_id]["metadata"] = {
-            k: (str(attrs[k], "utf-8") if isinstance(attrs[k], bytes) else attrs[k])
-            for k in ("chemistry_description", "software_version")
-        }
+        adata.uns["spatial"][library_id]["metadata"] = dict()
+        for k in ("chemistry_description", "software_version"):
+            try:
+                if isinstance(attrs[k], bytes):
+                    adata.uns["spatial"][library_id]["metadata"][k] = str(
+                        attrs[k], "utf-8"
+                    )
+                # this statement is necessary for the subsampled dataset in tests
+                elif isinstance(attrs[k], np.ndarray):
+                    adata.uns["spatial"][library_id]["metadata"][k] = str(
+                        attrs[k][0], "utf-8"
+                    )
+                else:
+                    adata.uns["spatial"][library_id]["metadata"][k] = attrs[k]
+            except Exception:
+                pass
```

Comment:
Could this just be:

```python
{
    k: (str(attrs[k], "utf-8") if isinstance(attrs[k], bytes) else attrs[k])
    for k in ("chemistry_description", "software_version") if k in attrs
}
```

Also, probably better to just add the attributes to the test data rather than change the code in the library.

Reply:
Done

Reply:
Thank you for the suggestion